src/extractor/guidance/intersection_handler.cpp

```diff
@@ -383,131 +394,147 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
-                                                     out_data.road_classification.GetPriority() &&
-                                                 in_classification != continue_class) ||
-                          (deviation < best_continue_deviation &&
-                           out_data.road_classification == continue_class) ||
-                          (continue_class != in_classification &&
-                           out_data.road_classification == continue_class)))
         {
-            best_continue_deviation = deviation;
-            best_continue = i;
+            out_ways.push_back(out);
+            continue;
         }
+        ++num_valid_entry;
 
-        const auto current_best_class =
-            node_based_graph.GetEdgeData(intersection[best_continue].eid).road_classification;
+        const auto out_node = node_based_graph.GetEdgeData(intersection[i].eid);
+        const auto out_classification = out_node.road_classification;
 
-        // don't prefer low priority classes
-        if (best != 0 && out_data.road_classification.IsLowPriorityRoadClass() &&
-            !current_best_class.IsLowPriorityRoadClass())
-            continue;
+        out.deviation_from_straight = angularDeviation(intersection[i].angle, STRAIGHT_ANGLE);
+        out.same_or_higher_priority =
+            out_node.road_classification.GetPriority() <= in_way.road_classification.GetPriority();
+        out.same_classification = out_classification == in_way.road_classification;
+        out.same_name_id = out_node.name_id == in_way.name_id;
 
-        const bool is_better_choice_by_priority =
-            best == 0 || obviousByRoadClass(in_data.road_classification,
-                                            out_data.road_classification,
-                                            current_best_class);
+        out_ways.push_back(out);
+    }
 
-        const bool other_is_better_choice_by_priority =
-            best != 0 && obviousByRoadClass(in_data.road_classification,
-                                            current_best_class,
-                                            out_data.road_classification);
+    const auto End = end(out_ways);
+    // dead end, or otherwise no exitable ways
+    if (num_valid_entry == 0)
+        return 0;
 
-        if ((!other_is_better_choice_by_priority && deviation < best_deviation) ||
-            is_better_choice_by_priority)
-        {
-            best_deviation = deviation;
-            best = i;
-        }
+    // sort out ways by what they share with the in way
+    // entry allowed, lowest deviation to greatest deviation, name, classification and priority
+    const auto sort_by = [](const out_way &lhs, const out_way &rhs) {
+        auto left_tie =
+            std::tie(lhs.same_classification, lhs.same_or_higher_priority);
+        auto right_tie =
+            std::tie(rhs.same_classification, rhs.same_or_higher_priority);
+        return lhs.entry_allowed > rhs.entry_allowed ||
```

Comment:
But you could still do:

`std::tie(a0,b1,a2) < std::tie(b0,a1,b2)` to achieve `a0 < a1`, if equal `a1 > a2`, if equal `a2 < b2`, right? After all `a0 < b0` is equivalent to `b0 > a0`. So all `>` and `<` can be switched by choosing the comparison side.